integration/spark-common/src/main/scala/org/apache/spark/sql/execution/streaming/CarbonAppendableStreamSink.scala

```diff
@@ -163,7 +162,7 @@ class CarbonAppendableStreamSink(
   /**
    * if the directory size of current segment beyond the threshold, hand off new segment
    */
-  private def checkOrHandOffSegment(): Unit = {
+  private def checkOrHandOffSegment(operationContext: OperationContext): Unit = {
```

Comment:
No need to change method signature use the same operation context passed in CarbonAppendableStreamSink